src/utils/event_loop.h

```diff
@@ -197,18 +198,21 @@ event_loop_uninitialize(struct event_loop *loop);
  * loop iteration.
  *
  * @param loop The event loop instance.
+ * @param retval The return value, 0 if success, an error code otherwise
```

Comment:
Based on its usage, it looks like you expect a negative errno value (which is good). You might want to specify this here, and possibly check for a negative value when assigning to `loop->exit_code`. Can even do: `loop->exit_code = min(exit_code, -exit_code)`.

Alternatively, you can require a positive errno value and don't worry about the sign. When grabbing this value later to return from `event_loop_run()`, you can verify/convert to negative as above with `min()`.